kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java

```diff
@@ -168,6 +170,24 @@ public void onClose(int i, String s) {
   public void cleanup() throws InterruptedException {
     client.pods().inNamespace(currentNamespace).withName(pod1.getMetadata().getName()).delete();
     // Wait for resources to get destroyed
-    Thread.sleep(30000);
+    await().atMost(2, TimeUnit.MINUTES).until(podDeleted());
+  }
+
+
+  private Callable<Boolean> podDeleted() {
+    return new Callable<Boolean>() {
+      public Boolean call() throws Exception {
+        return client.pods().inNamespace(currentNamespace).list().getItems().size() == 0; // The condition that must be fulfilled
+      }
+    };
+  }
+
+  private Callable<Boolean> podReady() {
```

Comment:
I will create a class called PodReady that implements `Callable`  and the constructor will receive a string that will be the pod name so they check if it ready. So instead of defining this logic in all tests, you can reuse it.